src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java

```diff
@@ -76,8 +76,14 @@ protected void doRun() throws Exception {
             long pending = queue.getPendingItems().size();
             long stuck = 0;
             long blocked = 0;
+            
             final Queue.Item[] items = queue.getItems();
             for (Queue.Item item : items) {
+                // Get the name of the task for job_name tag
+                Queue.Task task = item.task;
+                String display_name = task.getFullDisplayName();
+                tags = TagsUtil.addTagToTags(tags, "job_name", display_name);
```

Comment:
TagsUtil is here to make sure we never have twice a tag with the same key. With that logic here, you'll just end up with the name of the last item in the queue as a tag.